### PR TITLE
Bump apache parent to 31, jdk to 17 and springboot to 3.2.3

### PR DESCRIPTION
**Pull request by walklown**
Upgrade all use cases in the '2-advance' directory to spring-boot 3.x, jdk 17. Effect 114 files. Passed all tests. If it is not allowed to affect so much, then I can also give up my PR. @chickenlj 

https://github.com/apache/dubbo/issues/13862

### CODE REVIEW

File: 1-basic/dubbo-samples-spring-boot-idl/pom.xml [hunk 1]
```diff
@@ -53,6 +53,13 @@
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>org.apache</groupId>
+                <artifactId>apache</artifactId>
+                <version>31</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
```
Comment (chickenlj) [inline 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: no need to add this dependency here?

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: 1-basic/dubbo-samples-spring-boot-idl/pom.xml (hunk 1 above)
Comment (walklown) [reply 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: This determines the version of the ‘org.apache.maven.plugins’ plugin. I understand that the versions of these plugins should be unified. Or do you suggest that the versions of the plug-ins should be defined in the pom?

Comment (chickenlj) [reply 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: I don't understand why we need to add this dependency when the following has already been set as the parent of the demo project:

```xml
<parent>
    <groupId>org.apache</groupId>
    <artifactId>apache</artifactId>
    <version>31</version>
    <relativePath/>
</parent>
```

Comment (walklown) [reply 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: 1、The 'root pom.xml' and the '1-basic/pom.xml' does not set this parent. Maybe in diffrent branch ?

2、The project structure of the current project is split. For example, 'root pom.xml' sets '1-basic/pom.xml' as module, but '1-basic/pom.xml' does not set 'root pom.xml' as parent. Therefore, the component versions in '1-basic/pom.xml' and subordinate pom.xml are not controlled by 'root pom.xml'.

And the parent of 'dubbo-samples-spring-boot-idl/dubbo-samples-spring-boot-idl-provider/pom.xml' is set to 'dubbo-samples-spring-boot-idl/pom.xml', so dubbo -samples-spring-boot-idl The component versions of this subproject are controlled by 'dubbo-samples-spring-boot-idl/pom.xml'.

Comment (walklown) [reply 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: Means 'apache -> dubbo-samples-spring-boot-idl -> dubbo-samples-spring-boot-idl-provider/consumer', not 'root -> 1-basic -> dubbo-samples-spring-boot-idl -> dubbo-samples-spring-boot-idl-provider/consumer'. 

'modules' tag only works for 'The Reactor' of maven.
https://maven.apache.org/guides/mini/guide-multiple-modules.html

Comment (walklown) [reply 1-basic/dubbo-samples-spring-boot-idl/pom.xml:62]: I think I've said it more clearly, and I have no idea why I didn't get a follow-up reply. Then I decided to give up this PR.

**Pull request closed without merging**